execution_engine/src/storage/db_store.rs

```diff
@@ -0,0 +1,148 @@
+const ROCKS_DB_ZSTD_MAX_DICT_BYTES: i32 = 256 * 1024;
+const ROCKS_DB_MAX_LEVEL_FILE_SIZE_BYTES: u64 = 512 * 1024 * 1024;
+const ROCKS_DB_MAX_OPEN_FILES: i32 = 768;
+const ROCKS_DB_ZSTD_COMPRESSION_LEVEL: i32 = 3; // Default compression level
+const ROCKS_DB_ZSTD_STRATEGY: i32 = 4; // 4: Lazy
+const ROCKS_DB_WINDOW_BITS: i32 = -14;
+
+/// Column family name for the v1 trie data store.
+const TRIE_V1_COLUMN_FAMILY: &str = "trie_v1_column";
+
+/// Column family name for tracking progress of data migration.
+const LMDB_MIGRATED_STATE_ROOTS_COLUMN_FAMILY: &str = "lmdb_migrated_state_roots_column";
+
+/// Working set column famliy.
+const WORKING_SET_COLUMN_FAMILY: &str = "working_set";
+
+const ACTIVE_COLUMN_FAMILIES: &[&str] = &[
+    TRIE_V1_COLUMN_FAMILY,
+    LMDB_MIGRATED_STATE_ROOTS_COLUMN_FAMILY,
+    WORKING_SET_COLUMN_FAMILY,
+];
+
+/// Generic data store backed by RocksDb.
+#[derive(Clone)]
+pub struct DbStore {
+    pub(crate) db: Arc<DBWithThreadMode<MultiThreaded>>,
+    pub(crate) path: PathBuf,
+}
+
+impl DbStore {
+    /// Create a new store backed by RocksDB.
+    pub fn new(path: impl AsRef<Path>) -> Result<Self, rocksdb::Error> {
+        Self::new_with_opts(path, Self::rocksdb_defaults())
+    }
+
+    /// Create a new store backed by RocksDB with the specified options.
+    pub fn new_with_opts(
+        path: impl AsRef<Path>,
+        rocksdb_opts: Options,
+    ) -> Result<Self, rocksdb::Error> {
+        let db = Arc::new(DBWithThreadMode::<MultiThreaded>::open_cf(
+            &rocksdb_opts,
+            path.as_ref(),
+            ACTIVE_COLUMN_FAMILIES.to_vec(),
+        )?);
+
+        Ok(DbStore {
+            db,
+            path: path.as_ref().to_path_buf(),
+        })
+    }
+
+    /// Trie V1 column family.
+    pub(crate) fn trie_column_family(&self) -> Result<Arc<BoundColumnFamily<'_>>, error::Error> {
+        self.db.cf_handle(TRIE_V1_COLUMN_FAMILY).ok_or_else(|| {
+            error::Error::UnableToOpenColumnFamily(TRIE_V1_COLUMN_FAMILY.to_string())
+        })
+    }
+
+    /// Column family tracking state roots migrated from lmdb (supports safely resuming if the node
+    /// were to be stopped during a migration).
+    pub(crate) fn lmdb_tries_migrated_column(
+        &self,
+    ) -> Result<Arc<BoundColumnFamily<'_>>, error::Error> {
+        self.db
+            .cf_handle(LMDB_MIGRATED_STATE_ROOTS_COLUMN_FAMILY)
+            .ok_or_else(|| {
+                error::Error::UnableToOpenColumnFamily(
+                    LMDB_MIGRATED_STATE_ROOTS_COLUMN_FAMILY.to_string(),
+                )
+            })
+    }
+
+    /// Working set column family.
+    pub fn working_set_column_family(&self) -> Result<Arc<BoundColumnFamily<'_>>, error::Error> {
+        self.db.cf_handle(WORKING_SET_COLUMN_FAMILY).ok_or_else(|| {
+            error::Error::UnableToOpenColumnFamily(WORKING_SET_COLUMN_FAMILY.to_string())
+        })
+    }
+
+    /// Trie store iterator.
```

Comment:
Can you include in the comment what it is iterating over? It's not iterating over elements in the trie store - trie nodes, is it?

Reply:
Yes, it is, however I intended to move this to `RocksDbTrieStore`. Done in [6c20f4a](https://github.com/casper-network/casper-node/pull/2999/commits/6c20f4a0f076a2a375dc7d0f332871c085825545)